app/routes/surveys/AddSurveyRoute.js

```diff
@@ -1,31 +1,85 @@
 import { connect } from 'react-redux';
 import { compose } from 'redux';
-import { addSurvey, deleteSurvey } from '../../actions/SurveyActions';
+import {
+  addSurvey,
+  deleteSurvey,
+  fetchTemplate
+} from '../../actions/SurveyActions';
 import SurveyEditor from './components/SurveyEditor/SurveyEditor';
 import { LoginPage } from 'app/components/LoginForm';
 import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
 import { push } from 'react-router-redux';
-import moment from 'moment-timezone';
 import { initialQuestion } from './components/SurveyEditor/SurveyEditor';
+import prepare from 'app/utils/prepare';
+import { selectSurveyTemplate } from 'app/reducers/surveys';
+import { fetchEvent } from 'app/actions/EventActions';
+import { selectEventById } from 'app/reducers/events';
+import { defaultActiveFrom } from './utils';
 
-const time = (hours, minutes) =>
-  moment()
-    .startOf('day')
-    .add({ hours, minutes })
-    .toISOString();
+const loadData = (props, dispatch) => {
+  const { template, event } = props.location.query;
+  if (event) {
+    return dispatch(fetchEvent(event)).then(
+      result =>
+        result.success &&
+        dispatch(fetchTemplate(result.payload.entities.events[event].eventType))
+    );
+  }
+  if (template) {
+    return dispatch(fetchTemplate(template));
+  }
+};
 
 const mapStateToProps = (state, props) => {
+  const notFetching = !state.surveys.fetching && !state.events.fetching;
+  const { templateType, event } = props.location.query;
+
+  const fullEvent = selectEventById(state, { eventId: event });
+
+  const template = selectSurveyTemplate(state, {
+    ...props,
+    templateType: templateType || fullEvent.eventType
+  });
+
+  const initialEvent = event
+    ? {
+        value: fullEvent.id,
+        label: fullEvent.title
+      }
+    : '';
+  const activeFrom = event ? fullEvent.endTime : defaultActiveFrom(12, 0);
+
+  let initialValues = null;
+  if (
+    !(
+      ((templateType || event) && !template) ||
```

Comment:
this seems painful, what's going on here?

Reply:
Yeah I know, I've been thinking about how to fix this. The purpose is to make sure to return `initialValues: null` if either the event or the template is being fetched. Without this check, the logic would use `event: undefined` and/or `template: undefined` to assume we were in a regular case without any url params. That was a problem, because once `initialValues` have been passed to redux-form as anything else than null, it didn't seem to automatically reinitialize when `initialValues` changed. I suppose `if (!notFetching) { ... }` would be equivalent, I've just had some bad experience with the `fetching` flag being inconsistent. I'll check if it works